core/src/translate/mod.rs

```diff
@@ -97,30 +97,72 @@ pub fn translate(sql_statement: &SqlStatement) -> Result<Statement> {
             comment,
             ..
         } => {
-            let columns = columns
+            let mut primary_key: Option<Vec<usize>> = None;
```

Comment:
the code changes here, could you try writing the code without using `mut` keyword?
I think that then we can simplify the codes a lot.
if you stuck at something, then i will support on it.

Reply:
While I agree that this code (and other similar points) is rather ugly, I must admit that I am not sure how to clean it up. Do you have any suggestions?

Reply:
sure, for this part, I will provide suggestions on it. please give me some time. you can focus on other points.

Reply:
Cleaned this up.